Lib/pathlib.py

```diff
@@ -194,25 +189,23 @@ def resolve(self, path, strict=False):
         s = str(path)
         if not s:
             return os.getcwd()
+        if strict:
```

Comment:
Lines 189-191 make a bad assumption that the current working directory (CWD) is a final path. The CWD is an absolute path but not a final path. The assignment should be something like `s = str(path) or '.'`. In practice it may never be an issue since `PurePath.__str__` assigns and returns `self._str = self._format_parsed_parts(self._drv, self._root, self._parts) or '.'`.

Reply:
> The CWD is an absolute path but not a final path.

TIL.

I like the `or '.'` approach since it avoids special-casing things, but want @pitrou to take a look since this is inherited from the initial pathlib implementation.